### PR TITLE
Trie Implementation Using Java Hashmap 

### LINKED ISSUE #2968: Implementation of Trie Data structure Using Java Hashmap
## 💥 Proposal

Hii, I am Prachi
I want to add Java program to perform insert and search operations on Trie Data Structure.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes and I'll follow the same.

I am GSSoC'21 participant and would like to contribute to this project.
Kindly assign me this issue.

### PR DESCRIPTION
**Pull request by pm0824**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
I have used Java Hashmap to implement Trie Insert and Search String Operations in efficient way.

### Checklist
- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Resolved issue #2968

### CODE REVIEW

File: Java/ds/Trie_with_Hashmap.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
+/* Trie is also called as 'Prefix Tree'. It is a tree data structure which is commonly used
+*  to store the strings/words. 
+*  The popoular applications of this data stucture are autocomplete and spellcheck.
+*/
+
+
```
Comment (ankitaggarwal23) [inline Java/ds/Trie_with_Hashmap.java]: -Use Multiline comments
- Remove Extra lines one line is enough.

File: Java/ds/Trie_with_Hashmap.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+
+//Class to define Trie node operations
```
Comment (ankitaggarwal23) [inline Java/ds/Trie_with_Hashmap.java]: Remove Extra lines one line is enough.

File: Java/ds/Trie_with_Hashmap.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        boolean isEnd = false;
+
+        TrieNode(boolean isEnd){
+            this.isEnd = isEnd;
+        }
+    }
+
+
+}
+
```
Comment (ankitaggarwal23) [inline Java/ds/Trie_with_Hashmap.java]: Remove Extra lines.

File: Java/ds/Trie_with_Hashmap.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+class Trie_with_Hashmap{
+   public static void main(String[] args) {
+
+    //Trie object
+    Trie head = new Trie();
+
```
Comment (ankitaggarwal23) [inline Java/ds/Trie_with_Hashmap.java]: Remove Extra lines and unnecessary comments.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces

**Comment by pm0824** [comment]
Thank you for the review. I have made necessary changes, kindly check once again.

File: Java/ds/Trie_with_Hashmap.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                break;                          
+            case 2 : 
+                //Search function returns 'true' if string is present in Trie
+                System.out.print("Enter string to search : ");
+                System.out.println("Search result : "+ head.search( scan.next() ));
+                break;                                          
+            default : 
+                System.out.println("Invalid choice \n ");
+                break;   
+            }
```
Comment (ankitaggarwal23) [inline Java/ds/Trie_with_Hashmap.java]: Kindly properly align your this part of code.

File: Java/ds/Trie_with_Hashmap.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+ 
+            System.out.print("\nDo you want to continue (Type y or n) : ");
+            ch = scan.next().charAt(0);   
+
```
Comment (ankitaggarwal23) [inline Java/ds/Trie_with_Hashmap.java]: Remove that extra line.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by pm0824** [comment]
Changes made as per the request. Please review.

**Review by ankitaggarwal23 — APPROVED** [review]

File: Java/ds/Trie_with_Hashmap.java [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+                System.out.print("Enter string to search : ");
+                System.out.println("Search result : "+ head.search( scan.next() ));
+                break;                                          
+            default : 
+                System.out.println("Invalid choice \n ");
+                break;   
+            }
+            System.out.print("\nDo you want to continue (Type y or n) : ");
+            ch = scan.next().charAt(0);                                    
+        } while (ch == 'Y'|| ch == 'y');
```
Comment (ankitaggarwal23) [inline Java/ds/Trie_with_Hashmap.java]: - Code is not in a proper format. Take care of your braces aligning.
Ex- See your switch case braces and also it collides with cases.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by pm0824** [comment]
I have referred [`java conventions`](https://www.oracle.com/java/technologies/javase/codeconventions-statements.html) and made the changes accordingly. Kindly check it now.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**